src/internal.rs

```diff
@@ -213,5 +197,6 @@ impl<'a> SplayTree<'a> {
         current.left.set(null.right.get());
         current.right.set(null.left.get());
         self.root = Some(current);
```

Comment:
BTW, one can move this assignment to the calling sites to make it more explicit, and so `splay` will become a static method.